tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/debugger_container_test.ts

```diff
@@ -0,0 +1,132 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+/**
+ * Unit tests for the Debugger Container.
+ */
+import {CommonModule} from '@angular/common';
+import {TestBed} from '@angular/core/testing';
+import {By} from '@angular/platform-browser';
+
+import {Store} from '@ngrx/store';
+import {provideMockStore, MockStore} from '@ngrx/store/testing';
+
+import {debuggerLoaded} from './actions';
+import {DebuggerComponent} from './debugger_component';
+import {DebuggerContainer} from './debugger_container';
+import {DataLoadState, State} from './store/debugger_types';
+import {createDebuggerState, createState} from './testing';
+import {AlertsModule} from './views/alerts/alerts_module';
+import {InactiveModule} from './views/inactive/inactive_module';
+
+/** @typehack */ import * as _typeHackStore from '@ngrx/store';
+
+describe('Debugger Container test', () => {
+  let store: MockStore<State>;
+  let dispatchSpy: jasmine.Spy;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [DebuggerComponent, DebuggerContainer],
+      imports: [AlertsModule, CommonModule, InactiveModule],
+      providers: [
+        provideMockStore({
+          initialState: createState(createDebuggerState()),
+        }),
+        DebuggerContainer,
+      ],
+    }).compileComponents();
+    store = TestBed.get(Store);
+    dispatchSpy = spyOn(store, 'dispatch');
+  });
+
+  it('renders debugger component initially with inactive component', () => {
+    const fixture = TestBed.createComponent(DebuggerContainer);
+    fixture.detectChanges();
+
+    const inactiveElement = fixture.debugElement.query(
+      By.css('tf-debugger-v2-inactive')
+    );
+    expect(inactiveElement).toBeTruthy();
+    const alertsElement = fixture.debugElement.query(
+      By.css('tf-debugger-v2-alerts')
+    );
+    expect(alertsElement).toBeNull();
+  });
+
+  it('rendering debugger component dispatches debuggeRunsRequested', () => {
+    const fixture = TestBed.createComponent(DebuggerContainer);
+    fixture.detectChanges();
+    expect(dispatchSpy).toHaveBeenCalledWith(debuggerLoaded());
+  });
+
+  it('updates the UI to hide inactive component when store has 1 run', () => {
+    const fixture = TestBed.createComponent(DebuggerContainer);
+    fixture.detectChanges();
+
+    store.setState(
```

Comment:
I see your point. I think this pattern (namely setting store state) is more aligned with the testing principle of "test behavior, not internal implementation". For one thing, it's more resistant against changes in how the details of the selectors change. So I prefer to keep the current pattern. (Also, as you may know, the component tests in tensorboard/webapp are based on this pattern, instead of the `overrideSelector()` pattern.